hexrd/ui/overlays/laue_diffraction.py

```diff
@@ -130,4 +133,31 @@ def polar_ranges(self, spots):
             # Put the first point at the end to complete the square
             corners.append(corners[0])
             ranges.append(corners)
+
         return ranges
+
+    @staticmethod
+    def range_data(range_corners, display_mode, panel):
+        # This function is only for raw and cartesian views
+        if not range_corners:
+            return []
+
+        # The range data is curved for raw and cartesian.
+        # Get more intermediate points so the data reflects this.
+        results = []
+        for corners in range_corners:
+            data = []
+            for i in range(len(corners) - 1):
+                tmp = np.linspace(corners[i], corners[i + 1])
+                data.extend(panel.angles_to_cart(tmp))
+
+            data = np.array(data)
+            if display_mode == 'raw':
+                data = panel.cartToPixel(data)
+                data[:, [0, 1]] = data[:, [1, 0]]
+            elif display_mode == 'cartesian':
```

Comment:
Lets extract out the display modes into constants

Reply:
Done